t/21_sles4sap_azure_cli.t

```diff
@@ -276,17 +276,31 @@ subtest '[az_vm_create] with no public IP' => sub {
 subtest '[az_vm_list]' => sub {
     my $azcli = Test::MockModule->new('sles4sap::azure_cli', no_auto => 1);
     my @calls;
-    $azcli->redefine(script_output => sub { push @calls, $_[0]; return '["Mirandolina","Truffaldino"]'; });
+    $azcli->redefine(script_output => sub {
+            push @calls, $_[0];
+            return '["Mirandolina","Truffaldino"]'; });
 
     my $res = az_vm_list(resource_group => 'Arlecchino', query => 'ZAMZAM');
 
     note("\n  -->  " . join("\n  -->  ", @calls));
     ok((any { /az vm list/ } @calls), 'Correct composition of the main command');
     ok((any { /-g Arlecchino/ } @calls), 'Correct composition of the -g argument');
-    ok((any { /--query.*ZAMZAM/ } @calls), 'Correct composition of the --query argument');
```

Comment:
make sure to remove this line, right? I don't find the related code to this line, and other than that, LGTM

Reply:
Corresponding change is https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/19606/files#diff-2b900fafc5f19c0adfc596d832d2c3fd8b3fc90b64ed8a2d2ad6680596f28924R623-R624

So before this pr, the `query` argument was mandatory.
Now it become optional so I decided to test it like:

1. in test `subtest '[az_vm_list]'` I do not care to the argument at all (and I'll change the test to also call the tested function without `query =>`)
2. in the next new test `subtest '[az_vm_list] query'` I check that specifying the argument result in value used on the az cli command composition

Reply:
thank you, it's clear :D